python/ucxx/_lib/libucxx.pyx

```diff
@@ -40,15 +41,48 @@ include "tag.pyx"
 logger = logging.getLogger("ucx")
 
 
-np.import_array()
+cdef class HostBufferAdapter:
+    """A simple adapter around HostBuffer implementing the buffer protocol"""
+    cdef Py_ssize_t _size
+    cdef void* _ptr
+    cdef Py_ssize_t[1] _shape
+    cdef Py_ssize_t[1] _strides
+    cdef Py_ssize_t _itemsize
+
+    @staticmethod
+    cdef _from_host_buffer(HostBuffer* host_buffer):
+        """Construct a new HostBufferAdapter from a HostBuffer.
+
+        This factory takes ownership of the input host_buffer's data, so
+        attempting to use the input after this function is called will result
+        in undefined behavior.
+        """
+        cdef HostBufferAdapter obj = HostBufferAdapter.__new__(HostBufferAdapter)
+        obj._size = host_buffer.getSize()
+        obj._ptr = host_buffer.release()
+        obj._shape = [obj._size]
+        obj._itemsize = sizeof(uint8_t)
+        obj._strides = [obj._itemsize]
+        return obj
+
+    def __getbuffer__(self, Py_buffer *buffer, int flags):
+        buffer.buf = self._ptr
+        buffer.format = 'c'
```

Comment:
```suggestion
        buffer.format = 'B'
```
randomly noticed the tests were failing, `c` translates to a Python bytes/NumPy string dtype of length 1. `B` translates to the unsigned byte it was before.

Reply:
Ah so this is a signed vs unsigned discrepancy?